lib/vicadmin/validate.go

```diff
@@ -209,14 +204,37 @@ func NewValidator(ctx context.Context, vch *config.VirtualContainerHostConfigSpe
 		v.DockerPort = fmt.Sprintf("%d", opts.DefaultTLSHTTPPort)
 	}
 
-	err = v.QueryDatastore(ctx, vch, sess)
+	err := v.QueryDatastore(ctx, vch, sess)
 	if err != nil {
 		log.Errorf("Had a problem querying the datastores: %s", err.Error())
 	}
 	v.QueryVCHStatus(vch, sess)
 	return v
 }
 
+// Obtain the VCH name from vsphere
+func VCHName(ctx context.Context, sess *session.Session) string {
+	defer trace.End(trace.Begin(""))
+
+	self, err := guest.GetSelf(ctx, sess)
+	if err != nil {
+		log.Errorf("Unable to get VCH name due to unknown self-reference: %s", err)
+		log.Infof("Setting the VCH name to %s", DefaultVCHName)
+		return DefaultVCHName
+	}
+
+	newVM := vm.NewVirtualMachineFromVM(ctx, sess, self)
+	vchName, err := newVM.Name(ctx)
+	if err != nil {
+		log.Errorf("Unable to get VCH name: %s", err)
+		log.Infof("Setting the VCH name to %s", DefaultVCHName)
+		return DefaultVCHName
+	}
+
+	log.Infof("Setting the VCH name to %s", vchName)
```

Comment:
I'll take the blame for the current func name (https://github.com/vmware/vic/pull/4134#discussion_r104963502) - it was initially named `GetVCHName`. I agree with @gigawhitlocks.